lib/Builtins.cpp

```diff
@@ -14,171 +14,335 @@
 
 #include "Builtins.h"
 
+#include <stdlib.h>
+#include <unordered_map>
+
 using namespace llvm;
+using namespace clspv;
+
+namespace {
+////////////////////////////////////////////////////////////////////////////////
+////  Convert Builtin function name to a Type enum
+////////////////////////////////////////////////////////////////////////////////
+Builtins::BuiltinType LookupBuiltinType(const std::string &name) {
+
+// Build static map of builtin function names
+#include "BuiltinsMap.inc"
```

Comment:
Good idea to separate this out into its own file.  It's a big table with its own edit/review criteria.  :-)